databricks-sdk-java/src/main/java/com/databricks/sdk/core/DatabricksConfig.java

```diff
@@ -669,4 +676,41 @@ public DatabricksConfig newWithWorkspaceHost(String host) {
                 "headerFactory"));
     return clone(fieldsToSkip).setHost(host);
   }
+
+  /**
+   * Sets a custom TokenCache implementation.
+   *
+   * @param tokenCache the TokenCache implementation to use
+   * @return this config instance
+   */
+  public DatabricksConfig setTokenCache(TokenCache tokenCache) {
+    this.tokenCache = tokenCache;
+    return this;
+  }
+
+  /**
+   * Gets the default OAuth redirect URL. If one is not provided explicitly, uses
+   * http://localhost:8080/callback
+   *
+   * @return The OAuth redirect URL to use
+   */
+  public String getEffectiveOAuthRedirectUrl() {
+    return redirectUrl != null ? redirectUrl : "http://localhost:8080/callback";
+  }
+
+  /**
+   * Gets the TokenCache instance for the current configuration.
+   *
+   * <p>If a custom TokenCache has been set, it will be returned. Otherwise, a SimpleFileTokenCache
+   * will be created based on the configuration properties.
+   *
+   * @return A TokenCache instance
+   */
+  public synchronized TokenCache getTokenCache() {
```

Comment:
Just talking with @renaudhartert-db. Is it possible to inline this token cache entirely into ExternalBrowserCredentialsProvider? We're trying to break apart DatabricksConfig and stop using it as a dependency injection mechanism. I think you could do this by
1. accepting the TokenCache as a constructor parameter in ExternalBrowserCredentialsProvider, falling back to this implementation if not specified
2. Adding the token cache to SessionCredentials via its builder.